bootstrap/pkg/utils/k8utils.go

```diff
@@ -279,6 +279,12 @@ func NewApply(namespace string, restConfig *rest.Config) (*Apply, error) {
 	if err != nil {
 		return nil, err
 	}
+
+	// Change behaviour of error from exit to panic.
+	cmdutil.BehaviorOnFatal(func(s string, i int) {
+		panic(fmt.Errorf("Encountered error: %s. Exiting with code: %d.", s, i))
```

Comment:
Why are we doing panic? Doesn't panic attempt to crash the program? Why would we want to crash?

Reply:
@jlewi the kubectl-related libraries use the cmdutil CheckErr function, which uses os.Exit by default.
By changing the exit to a panic, we are given the chance to intercept the panic and return an error instead.

Reply:
@krishnadurai How is that different from the default behavior which is os.Exit? If there is some place in the kfctl code base that is calling a kubectl library (e.g. cmdutil) then if it returns an error by default it would just call os.Exit and terminate execution if I'm understanding @yanniszark correctly.

So if we change the default behavior to panic and don't intercept the panic then the behavior would still be to crash on error.

Reply:
@jlewi behaviour wise things would be the same for panic and os.Exit. The only difference is what's printed out to stdout. If there's an unhandled panic it would lead to kfctl printing a traceback like the one shown below (simulated this one):

```
panic: Encountered error: error: no objects passed to apply. Exiting with code: 1.

goroutine 1 [running]:
github.com/kubeflow/kubeflow/bootstrap/v3/pkg/utils.NewApply.func1(0xc00067fec0, 0x21, 0x1)
	/Users/duselvar/golang/src/github.com/kubeflow/kubeflow/bootstrap/pkg/utils/k8utils.go:285 +0x12e
k8s.io/kubernetes/pkg/kubectl/cmd/util.checkErr(0x5b65700, 0xc000f18130, 0x58fb7b8)
	/Users/duselvar/golang/pkg/mod/k8s.io/kubernetes@v1.13.4/pkg/kubectl/cmd/util/helpers.go:170 +0x8da
k8s.io/kubernetes/pkg/kubectl/cmd/util.CheckErr(...)
	/Users/duselvar/golang/pkg/mod/k8s.io/kubernetes@v1.13.4/pkg/kubectl/cmd/util/helpers.go:116
github.com/kubeflow/kubeflow/bootstrap/v3/pkg/utils.(*Apply).run(0xc0004b89c0, 0x0, 0x0)
	/Users/duselvar/golang/src/github.com/kubeflow/kubeflow/bootstrap/pkg/utils/k8utils.go:335 +0x62
...
```